### DISCUSSION TITLE
Newly added worker node gets localhost hostname

### DISCUSSION THREAD

**Discussion by dkoci** [okd-project/okd#1039] [Q&A] [answered]
Hi.
I have added new worker node in my okd 4.8 cluster. Newly added worker node gets localhost hostname.
My cluster is running on vmware, UPI installed.
Is there any way to update the hostname?

**Comment by rvanderp3** [comment, marked as answer]
The new node should have an a PTR record which the node will use to resolve it's hostname.  

Alternatively, you can:
- `oc delete node localhost` 
- ssh to the the node and set the host name with hostnamectl --static
- restart the node

After the node restarts, it should rejoin with the intended node name.

**Reply by dkoci** [reply]
It works. Thanks alot.
Btw, what is a PTR record and how to add it to the new node?

**Reply by rvanderp3** [reply]
Awesome! I'm glad you got that working.  A PTR record is a type of DNS record which associates an IP with a domain name.   This enables reverse lookup of an IP to determine its associated domain name.  This is one of the ways the node determines its hostname, and by extension, its node name.  Another method by which a node name can be set is by providing the hostname via DHCP.  See [Setting the cluster node hostnames through DHCP](https://docs.okd.io/latest/installing/installing_bare_metal/installing-bare-metal.html#installation-host-names-dhcp-user-infra_installing-bare-metal) for additional doc on how the hostname is determined for a given node.

**Comment by LorbusChris** [comment]
~~This might be caused by [BZ192](https://bugzilla.redhat.com/show_bug.cgi?id=1929160)
Proposed fix: https://github.com/openshift/machine-config-operator/pull/2488~~
Edit: Nevermind, that's actually a bare metal issue
